mergify_engine/actions/merge/queue.py

```diff
@@ -76,30 +76,37 @@ def get_merge_method(
         """
         return self.redis.get(self._method_cache_key(pull_number)) or default
 
-    def _add_pull(self, pull_number, update=False):
+    def _add_pull(self, pull_number, priority, update=False):
         """Add a pull without setting its method.
 
         :param update: If update is True, don't create PR if it's not there.
         """
-        score = utils.utcnow().timestamp()
+        score = utils.utcnow().timestamp() / (priority + 1)
```

Comment:
Hm that's not perfect but I guess it's good enough. :)

Reply:
Well, since we use bigger numbers I can put `1 <= priority <= 10000` and remove the `+1`?
Or you're thinking of something else?

Reply:
No, I was thinking that in *theory* the worst case is that both score are equals, e.g. timestamp = 1 prio = 1 and then score = 1, timestamp =2 prio = 2 score = 1.

Not a big deal.